### PR TITLE
feat(FR-35): add a refresh button to the session launcher environment selector

### LINKED ISSUE #2906: Add Refresh Button to Environment Selector in Session Launcher
### Description

The Environment (image) selector in the session launcher currently fetches the list of images when the page is first loaded. However, images may be added asynchronously, such as through user image commit functions. To accommodate this, we should add a refresh button to the Environment selector. This will allow users to manually refresh the list of available images to ensure they have access to the most up-to-date options.

#### Steps to Implement

1. Add a refresh button next to the Environment selector in the session launcher.
1. Implement functionality to re-fetch the list of available images when the refresh button is clicked.
1. Ensure that the UI updates to reflect any new images that have been added asynchronously.
   #### Additional Context

- This feature is necessary to handle cases where images are added after the initial page load.
- Consider any potential performance impacts of re-fetching the image list.

JIRA Issue: FR-35

### PR DESCRIPTION
**Pull request by yomybaby**
Resolves #2906 (FR-35)

The session launcher's Environment / Version selector fetched the image list
once on mount (`useLazyLoadQuery` with `store-and-network` and no fetch key).
An image committed or pushed while the launcher was open only became selectable
after a full page reload.

## What changed

- `ImageEnvironmentSelectFormItems` gains an opt-in `showRefreshButton` prop.
  When set, a ghost refresh `IconButton` renders in the Environment / Version
  label row, next to the existing copy control.
- Clicking it bumps a `useUpdatableState` fetch key inside a `useTransition`,
  so the query re-runs with `network-only` while the already-rendered selects
  stay on screen instead of suspending the launcher step. The initial mount
  keeps `store-and-network`.
- `SessionLauncherPage` passes the new prop.

## Design decisions

- **Opt-in, not always on.** The component is also mounted by
  `DeploymentAddRevisionModal`; that call site is left untouched, so the button
  appears only where the issue asks for it.
- **`preventDefault()` on the click.** The label row is rendered inside the form
  item's `<label htmlFor>`, whose default action moves focus into the select —
  the same reason the form engine's own tooltip trigger prevents it.
- **No new i18n key.** The existing `button.Refresh` (en/ko and the other 20
  locales) covers the label and the tooltip, so `make i18n` was not needed.
- **The auto-select effect is untouched.** It keys off
  `environments.version` / `manual`, not the image list, so a refresh does not
  rewrite or reset the user's current selection.

## Tests

- New `react/src/components/ImageEnvironmentSelectFormItems.test.tsx` (2 cases,
  relay-test-utils mock environment): the button is absent without the prop, and
  clicking it puts a second `ImageEnvironmentSelectFormItemsQuery` in flight.
  `pnpm exec vitest run src/components/ImageEnvironmentSelectFormItems.test.tsx`
  → 2 passed.
- No e2e run (requires a live cluster).

## Verification

`bash scripts/verify.sh` →

```
=== ALL PASS ===
```

## Review notes

- Open the session launcher, reach the Environment step, and confirm the refresh
  icon sits to the right of the "Environments / Version" label.
- Commit a new customized image (or `docker push` one) with the launcher open,
  click refresh, and confirm the new image shows up in the Environment list
  without reloading the page.
- With an environment/version already picked, click refresh and confirm the
  selection is preserved and the step does not flash a loading fallback.
- Open Deployments → Add revision and confirm that selector still has **no**
  refresh button.

**Checklist:** (if applicable)

- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [x] Test case(s) to demonstrate the difference of before/after

https://claude.ai/code/session_011RFmSEBxxxvCXyquSPtqVJ

### CODE REVIEW

File: react/src/components/ImageEnvironmentSelectFormItems.test.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    // The initial query is already resolved, so nothing is in flight.
+    expect(environment.mock.getAllOperations()).toHaveLength(0);
+
+    await userEvent.click(refreshButton);
+
+    await waitFor(() => {
+      expect(environment.mock.getAllOperations()).toHaveLength(1);
+    });
+    expect(environment.mock.getMostRecentOperation().fragment.node.name).toBe(
+      'ImageEnvironmentSelectFormItemsQuery',
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply react/src/components/ImageEnvironmentSelectFormItems.test.tsx:150]: Fixed in `984e8d73a` — the refresh test now resolves the second operation instead of stopping at "a request started". The initial payload returns `pytorch` only, the refreshed payload returns `pytorch` + `tensorflow` (deterministic `MockPayloadGenerator` resolvers replace the generated one so the option names are readable), and the test opens the environment dropdown and asserts both options are offered — with a pre-refresh assertion that `tensorflow` is absent, so a regression that re-queries but keeps the selector on stale data now fails.

`bash scripts/verify.sh` → `=== ALL PASS ===`; `vitest run src/components/ImageEnvironmentSelectFormItems.test.tsx` → 2 passed.

**Comment by yomybaby** [comment]
Addressed Copilot's second-pass finding (review body, no inline thread — "Assert that the fallback is absent while the second operation remains in flight").

Fixed in `d5c1722a9`: between starting the refetch and resolving it, the test now asserts the Suspense fallback is absent and the environment trigger still renders, so a regression that dropped the `useTransition` wrapper — suspending the launcher step on every refresh — fails instead of passing.

Verified by mutation: replacing `startRefetchTransition(() => updateImageFetchKey())` with a bare `updateImageFetchKey()` makes the new assertion fail, so it is not vacuous.

`bash scripts/verify.sh` → `=== ALL PASS ===`.

The remaining Moderate finding (refresh-failure recovery, `ImageEnvironmentSelectFormItems.tsx:191`) is deliberately left open for a human decision — see that thread.
